File: gremlin-java/src/test/java/com/tinkerpop/gremlin/java/filter/HasStepTest.java
package com.tinkerpop.gremlin.java.filter;

import com.tinkerpop.blueprints.Graph;
import com.tinkerpop.blueprints.impls.tg.TinkerGraphFactory;
import com.tinkerpop.gremlin.Tokens.T;
import com.tinkerpop.gremlin.java.GremlinPipeline;
import com.tinkerpop.gremlin.test.ComplianceTest;

import java.util.Arrays;

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */
public class HasStepTest extends com.tinkerpop.gremlin.test.filter.HasStepTest {

    Graph g = TinkerGraphFactory.createTinkerGraph();

    public void testCompliance() {
        ComplianceTest.testCompliance(this.getClass());
    }

    public void test_g_V_hasXname_markoX() {
        super.test_g_V_hasXname_markoX(new GremlinPipeline(g).V().has("name", "marko"));
        super.test_g_V_hasXname_markoX(new GremlinPipeline(g.getVertices()).optimize(false).has("name", "marko"));
    }

    public void test_g_V_hasXname_blahX() {
        super.test_g_V_hasXname_blahX(new GremlinPipeline(g.getVertices()).has("name", "blah"));
        super.test_g_V_hasXname_blahX(new GremlinPipeline(g.getVertices()).optimize(false).has("name", "blah"));
    }

    public void test_g_V_hasXblahX() {
        super.test_g_V_hasXblahX(new GremlinPipeline(g.getVertices()).has("blah"));
        super.test_g_V_hasXblahX(new GremlinPipeline(g).V().optimize(false).has("blah"));
    }

    public void test_g_v1_out_hasXid_2X() {
        super.test_g_v1_out_hasXid_2X(new GremlinPipeline(g.getVertex(1)).out().has("id", "2"));
        super.test_g_v1_out_hasXid_2X(new GremlinPipeline(g.getVertex(1)).optimize(false).out().has("id", "2"));
    }

    public void test_g_V_hasXage_gt_30X() {
        super.test_g_V_hasXage_gt_30X(new GremlinPipeline(g).V().has("age", T.gt, 30));
        super.test_g_V_hasXage_gt_30X(new GremlinPipeline(g).V().optimize(false).has("age", T.gt, 30));
    }

    public void test_g_E_hasXlabelXknowsX() {
        super.test_g_E_hasXlabelXknowsX(new GremlinPipeline(g).E().has("label", T.eq, "knows"));
        super.test_g_E_hasXlabelXknowsX(new GremlinPipeline(g).optimize(false).E().has("label", T.eq, "knows"));
    }

    public void test_g_E_hasXlabelXknows_createdX() {
        super.test_g_E_hasXlabelXknows_createdX(new GremlinPipeline(g.getEdges()).has("label", T.in, Arrays.asList("knows", "created")));
        super.test_g_E_hasXlabelXknows_createdX(new GremlinPipeline(g).E().optimize(false).has("label", T.in, Arrays.asList("knows", "created")));

    }
}

File: gremlin-scala/src/test/scala/com/tinkerpop/gremlin/scala/filter/HasStepTest.scala
package com.tinkerpop.gremlin.scala.filter

import com.tinkerpop.blueprints.pgm.impls.tg.TinkerGraphFactory
import com.tinkerpop.gremlin.test.ComplianceTest
import com.tinkerpop.gremlin.scala._
import com.tinkerpop.gremlin.Tokens.T

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */

class HasStepTest extends com.tinkerpop.gremlin.test.filter.HasStepTest {

  val g = TinkerGraphFactory.createTinkerGraph();

  override def testCompliance() {
    ComplianceTest.testCompliance(this.getClass)
  }

  def test_g_V_hasXname_markoX() {
    super.test_g_V_hasXname_markoX(g.V.has("name", "marko"));
  }

  def test_g_V_hasXname_blahX() {
    super.test_g_V_hasXname_blahX(g.V.has("name", "blah"));
  }

  def test_g_V_hasXblah_nullX() {
    super.test_g_V_hasXblah_nullX(g.V.has("blah", null));
  }

  def test_g_v1_out_hasXid_2X() {
    super.test_g_v1_out_hasXid_2X(g.v(1).out.has("id", "2"));
  }

  def test_g_V_hasXage_gt_30X() {
    super.test_g_V_hasXage_gt_30X(g.V.has("age", T.gt, 30));
  }
}
